src/app/sessions/effects/session.effects.ts

```diff
@@ -1,23 +1,23 @@
 import { Injectable } from '@angular/core';
 import { Actions, Effect, ofType } from '@ngrx/effects';
 import { Observable } from 'rxjs/Observable';
-import { catchError, map, mergeMap, retryWhen, concatMap, tap } from 'rxjs/operators';
+import { catchError, map, mergeMap, retryWhen, concatMap, tap, withLatestFrom, distinctUntilChanged } from 'rxjs/operators';
 import { of } from 'rxjs/observable/of';
-import { Action } from '@ngrx/store';
+import { Action, Store } from '@ngrx/store';
 import * as sessionActions from '../actions/session.action';
-import * as sessionCreationActions from '../actions/session-creation.action';
+import * as sessionTransactionActions from '../actions/session-transaction.action';
 import * as roomActions from '../../rooms/actions/room.action';
 import * as problemActions from '../../problems/actions/problem.action';
+import * as fromSessionIndex from '../reducers/index';
 import * as judgeActions from '../../judges/actions/judge.action';
 import * as hearingPartsActions from '../../hearing-part/actions/hearing-part.action';
 import { SessionsService } from '../services/sessions-service';
 import { HttpErrorResponse } from '@angular/common/http';
 import { SearchFailed, SessionActionTypes } from '../actions/session.action';
 import 'rxjs/add/observable/timer';
 import 'rxjs/add/operator/mergeMap';
-import { GetProblemsForSession } from '../actions/session-creation.action';
 import { ProblemsService } from '../../problems/services/problems.service';
-import { Session } from '../models/session.model';
+import { RulesProcessingStatuses, TransactionBackendService, TransactionStatuses } from '../../core/services/transaction-backend.service';
 
 @Injectable()
 export class SessionEffects {
```

Comment:
This file scares me, as I'm starting to loose the ida of what's going on, but it doesn't seem to be wrong at the same time.